### PR TITLE
refactor!: remove deprecated entities

### PR DESCRIPTION
**Pull request by vladimirpotekhin**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes # 

## Does this PR introduce a breaking change?

BREAKING CHANGES: remove WithDateMaskPipeConfig, TUI_DATE_MASK, TUI_DATE_RANGE_MASK, TuiProgressColorSegmentsAsyncPipe, TuiNotificationService, TuiProgressColorSegmentsPipe, COUNTRIES, Country, second argument in TuiHideSelectedPipe

- [x] Yes
- [ ] No

### CODE REVIEW

File: projects/core/utils/dom/are-css-vars-supported.ts [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
  * @deprecated: use {@link tuiAreCssVarsSupported} instead
  * TODO: 3.0 remove this function
  */
 // eslint-disable-next-line @typescript-eslint/naming-convention
 export function areCssVarsSupported(cssRef: typeof CSS, userAgent: string): boolean {
-    return cssRef.supports('(--a: 0)') && !isEdgeOlderThan(17, userAgent);
+    return cssRef.supports('(--a: 0)') && !tuiIsEdgeOlderThan(17, userAgent);
 }
 
 export const tuiAreCssVarsSupported = areCssVarsSupported;
```
Comment (waterplea) [inline projects/core/utils/dom/are-css-vars-supported.ts:12]: Forgot this one

File: projects/kit/components/input-count/input-count-options.ts [hunk 2]
```diff
@@ -14,12 +14,6 @@ export interface TuiInputCountOptions {
     readonly postfix: string;
 }
 
-/**
- * @deprecated: use TuiInputCountOptions instead
- * todo: remove in 3.0
- */
-export type InputCountOptions = TuiInputCountOptions;
-
 // TODO: 3.0 remove in ivy compilation
```
Comment (waterplea) [inline projects/kit/components/input-count/input-count-options.ts:17]: Gotta not forget to go over all these fellas :D

**Review by waterplea — APPROVED** [review]

File: projects/kit/components/input-count/input-count-options.ts (hunk 2 above)
Comment (vladimirpotekhin) [reply projects/kit/components/input-count/input-count-options.ts:17]: yes)) I'll to go over all related with "ivy compilation" after ng12 update

File: projects/kit/components/input-phone-international/tokens/countries-masks.ts [hunk 3]
```diff
@@ -1,9 +1,215 @@
 import {InjectionToken} from '@angular/core';
 import {TuiCountryIsoCode} from '@taiga-ui/i18n';
 
-import {COUNTRIES_MASKS} from '../const/countries';
-
 export const TUI_COUNTRIES_MASKS: InjectionToken<Record<TuiCountryIsoCode, string>> =
     new InjectionToken<Record<TuiCountryIsoCode, string>>('Countries masks', {
-        factory: () => COUNTRIES_MASKS,
+        factory: () => ({
+            [TuiCountryIsoCode.AD]: '+376###-###',
```
[the remark this answers is not in the document]
Comment (vladimirpotekhin) [reply projects/kit/components/input-phone-international/tokens/countries-masks.ts:7]: Good idea, should we create repo in tinkoff space or completely separate ?

**Pull request merged**